qiskit/extensions/standard/u1.py

```diff
@@ -26,15 +23,8 @@ def __init__(self, theta, qubit, circ=None):
         super().__init__("u1", [theta], [qubit], circ)
 
     def _define_decompositions(self):
-        decomposition = DAGCircuit()
-        q = QuantumRegister(1, "q")
-        decomposition.add_qreg(q)
-        rule = [
-            UBase(0, 0, self.params[0], q[0])
-        ]
-        for inst in rule:
-            decomposition.apply_operation_back(inst)
-        self._decompositions = [decomposition]
+        """No decomposition as U is not a valid basis."""
```

Comment:
I'm not sure if "U is not a valid basis" is a correct statement... Maybe remove these comments?

Reply:
I believe that's what Jay was meaning here: https://github.com/Qiskit/qiskit-terra/issues/1697#issuecomment-456909795.

Reply:
I think what he meant is that there is nothing special about U and CX. The basis can be `['U','CX']` as any other list of gates. @jaygambetta ?

Reply:
Let's keep 'U' and 'CX' as valid basis - sorry for the confusion (spoke with @jaygambetta, and he meant those should not get enforced as default basis, which is what you are doing in this PR anyway).

Reply:
@ajavadia to be clear, do you mean to revert `_define_decompositions` for `u1`, `u2`, `u3`, and `cx` to what they were previously, so that they do return meaningful and non-null decompositions?

Reply:
yes exactly.